### PR TITLE
Refactor/remove transaction invocation

### PR DESCRIPTION
**Pull request by iamyulong**
## Summary

This PR removes the special `TransactionProcessorRunInvocation` in favor of `call_function`/`call_method`.

## Details

- Removed `TransactionProcessorRunInvocation`
- Added `(Local)Reference` to Scrypto value model

### CODE REVIEW

File: radix-engine-interface/src/constants.rs [hunk 1]
```diff
@@ -18,19 +18,41 @@ pub const PACKAGE_TOKEN: ResourceAddress = vanity_address!(EntityType::Resource,
 pub const OLYMPIA_VALIDATOR_TOKEN: ResourceAddress = vanity_address!(EntityType::Resource, 5);
 
 /// The address of the faucet package.
-pub const PACKAGE_LOADER: PackageAddress = vanity_address!(EntityType::Package, 0);
+pub const PACKAGE_PACKAGE: PackageAddress = vanity_address!(EntityType::Package, 0);
```
Comment (talekhinezh) [inline radix-engine-interface/src/constants.rs:21]: Package Package!

File: radix-engine/src/blueprints/transaction_processor/package.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+use super::TransactionProcessorBlueprint;
+
+pub struct TransactionProcessorNativePackage;
+
+impl TransactionProcessorNativePackage {
+    pub fn schema() -> PackageSchema {
+        let mut aggregator = TypeAggregator::<ScryptoCustomTypeKind>::new();
+
+        let mut substates = Vec::new();
+        substates.push(aggregator.add_child_type_and_descendents::<()>());
```
Comment (talekhinezh) [inline radix-engine/src/blueprints/transaction_processor/package.rs:24]: I didn't question this in the last PR but why do you think we should have atleast one substate per blueprint?

File: radix-engine-common/src/data/scrypto/custom_schema.rs [hunk 3]
```diff
@@ -24,6 +24,8 @@ pub enum ScryptoCustomTypeKind {
     Decimal,
     PreciseDecimal,
     NonFungibleLocalId,
+
+    Reference,
```
Comment (talekhinezh) [inline radix-engine-common/src/data/scrypto/custom_schema.rs:28]: Can we rename this to `InternalRef` since `Address` is a reference in itself as well.

**Review by talekhinezh — COMMENTED** [review]
Maybe I missed it but do we have any checks to prevent `Reference` from being stored in any substate?

File: radix-engine/src/blueprints/transaction_processor/package.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/blueprints/transaction_processor/package.rs:24]: It's changed in later PRs. `SELF` substates can be empty.

The original motivation of making `[0]` mandatory is for state-based access rules.

Changed auth to tolerate missing `[0]` as well.

File: radix-engine-common/src/data/scrypto/custom_schema.rs (hunk 3 above)
Comment (iamyulong) [reply radix-engine-common/src/data/scrypto/custom_schema.rs:28]: Sure

**Comment by iamyulong** [comment]
Good point about reference in storage!

I originally planed to do that as part of the reference overhaul story, but let me add some basic check.

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**